include/bluetooth/audio/cap.h

```diff
@@ -86,6 +137,210 @@ struct bt_cap_cb {
  */
 int bt_cap_find_ccid_service_service(struct bt_conn *conn, uint8_t ccid);
 
+/**
+ * @brief Discovers audio support on a remote device.
+ *
+ * This will discover the Common Audio Service (CAS) on the remote device, to
+ * if the remote device supports the Common Audio Profile.
+ *
+ * @param conn Connection to a remote server.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_cap_unicast_discover(struct bt_conn *conn);
+
+/** Type of CAP set */
+enum bt_cap_set_type {
+	/** The set is an ad-hoc set */
+	BT_CAP_SET_TYPE_AD_HOC,
+	/** The set is a CSIP Coordinated Set */
+	BT_CAP_SET_TYPE_CSIP,
+};
+
+struct bt_cap_unicast_audio_start_param {
+	/** The type of the set */
+	enum bt_cap_set_type type;
+
+	/** The number of set members in @p members and number of streams in @p streams */
+	uint8_t count;
+
+	union {
+		/**
+		 * @brief List of connection pointers if @p type is BT_CAP_SET_TYPE_AD_HOC
+		 *
+		 * The number of pointers in this array shall be at least @p count
+		 */
+		struct bt_conn **members;
+
+		/** CSIP Coordinated Set struct used if @p type is BT_CAP_SET_TYPE_CSIP */
+		struct {
+			/**
+			 * @brief CSIP Set Members in the set
+			 *
+			 * The number of pointers in this array shall be at least %p count
+			 */
+			struct bt_csis_client_set_member **members;
+
+			/** Set information used to verify the set @p members */
+			struct bt_csis_client_set_info *set_info;
+		} csip;
+	};
+
+	/** @brief Streams for the @p members
+	 *
+	 * stream[i] will be associated with members[i] if not already
+	 * initialized, else the stream will be verified against the member.
+	 */
+	struct bt_audio_stream **streams;
```

Comment:
Could you put a light  on how this is going to be used?  In what state `stream` is provided here?

Also I see that `struct bt_audio_stream` containst `struct bt_conn` and `struct bt_audio_ep`, so maybe in the end we don't need the above union at all as list of streams actually defines the group?

Reply:
> In what state stream is provided here?

Can you elaborate on what you mean by `state` here? 

> Also I see that struct bt_audio_stream containst struct bt_conn and struct bt_audio_ep, so maybe in the end we don't need the above union at all as list of streams actually defines the group?

`struct bt_audio_stream` is application allocated and does not (or should not) have any application-assigned values for `bt_conn` or `bt_audio_ep` (that is not very clear from the API as of now, I agree). This is probably originally been designed based on the `l2cap` API, which uses the same formula so to say (which possibly isn't a very good one, since it is easy for the application to modify these values and cause issues). 

So `struct bt_audio_stream` contains some applicaiton-assigned values, and then a couple of values that the stack, i.e. CAP/BAP will fill in, such as the `bt_conn` and `bt_audio_ep`.

Reply:
Need to agree this is confusing. For example, how user can say create request for following usecase
 - 1 Sink stream to Device A
 - 1 Source stream to Device A
 - 1 Sink stream to Device B
 ?
All in one CIG of course.

Reply:
Hmm, that is a good point. 

To achieve that the application would need to do
```
members[0] = device_1;
streams[0] = device_1_sink_stream;
members[0] = device_1;
streams[0] = device_1_source_stream;
members[0] = device_2;
streams[0] = device_2_sink_stream;
```

and then `count` should mainly refer to the number of streams. 

Might need to rethink this to see if there's a better way to avoid multiple 

The alternative is to couple the streams and member pointers in a preliminary function, and then just provide the list of streams here. 

Please let me know if you have any suggestions :)

Reply:
I think  I would suggest that CAP has high level API here. Eg.
1) User provides to CAP list of members which are involved in the scenario. (as it is now)
2) User provides to CAP information what scenario should be realized by CAP.  E.g number of Sink connection per device and number of source connections. Or maybe even just a stream context type which would match to some policy inside the CAP e.g. 
    a) for MEDIA, CAP will establish uni direction CISes to all the devices in *members 
    b) For Phone, CAP will  establish bi-directional CISes to all the devices in *members.
3) user could provide target latency and required sample frequency to Sink and Source streams.
4) CAP would take care for allocation memory for `struct bt_audio_stream`  per device (not sure if possible) while creating streams

Reply:
> 2. User provides to CAP information what scenario should be realized by CAP

I think it makes sense to use the codec config data and metadata to retrieve as much information about the scenario as possible. The details is not something I've considered too much yet. 


> 3. user could provide target latency and required sample frequency to Sink and Source streams.

Agree, that will be part of the `qos` parameter here

> 4. CAP would take care for allocation memory for struct bt_audio_stream per device (not sure if possible) while creating streams

This one I am not so sure about. I am not opposed to allocating the streams in CAP, but I also do not see the benefit. Can you elaborate on what your thoughts here are?